src/lib/connectors/magicConnector.ts

```diff
@@ -0,0 +1,124 @@
+import { ConnectExtension } from '@magic-ext/connect';
+import { OAuthExtension, OAuthProvider } from '@magic-ext/oauth';
+import { InstanceWithExtensions, SDKBase } from '@magic-sdk/provider';
+import { RPCProviderModule } from '@magic-sdk/provider/dist/types/modules/rpc-provider';
+import { Chain, Connector, normalizeChainId } from '@wagmi/core';
+import { ethers, Signer } from 'ethers';
+import { getAddress } from 'ethers/lib/utils';
+import { AbstractProvider } from 'web3-core';
+
+import { createModal } from '../modal/view';
+
+const IS_SERVER = typeof window === 'undefined';
+
+export interface MagicOptions {
+  apiKey: string;
+  accentColor?: string;
+  isDarkMode?: boolean;
+  customLogo?: string;
+  customHeaderText?: string;
+}
+
+interface UserDetails {
+  email: string;
+  phoneNumber: string;
+  oauthProvider: OAuthProvider;
+}
+
+export abstract class MagicConnector extends Connector {
+  ready = !IS_SERVER;
+
+  readonly id = 'magic';
+
+  readonly name = 'Magic';
+
+  provider: RPCProviderModule & AbstractProvider;
+
+  isModalOpen = false;
+
+  magicOptions: MagicOptions;
+
+  protected constructor(config: { chains?: Chain[]; options: MagicOptions }) {
+    super(config);
+    this.magicOptions = config.options;
+  }
+
+  async getAccount(): Promise<string> {
+    const provider = new ethers.providers.Web3Provider(
+      await this.getProvider()
+    );
+    const signer = provider.getSigner();
+    const account = await signer.getAddress();
+    return account;
+  }
+
+  async getUserDetailsByForm(
+    enableSMSLogin: boolean,
+    enableEmailLogin: boolean,
+    oauthProviders: OAuthProvider[]
+  ): Promise<UserDetails> {
+    const output: UserDetails = (await createModal({
+      accentColor: this.magicOptions.accentColor,
+      isDarkMode: this.magicOptions.isDarkMode,
+      customLogo: this.magicOptions.customLogo,
+      customHeaderText: this.magicOptions.customHeaderText,
+      enableSMSLogin: enableSMSLogin,
+      enableEmailLogin: enableEmailLogin,
```

Comment:
```suggestion
      enableEmailLogin: enableEmailLogin || true,
```
This makes the email login to be enabled by default.